File: backend/app/schemas/management.py
from decimal import Decimal

from pydantic import BaseModel


class BranchRankingItemResponse(BaseModel):
    branch_id: int
    branch_name: str
    branch_code: str
    total_disbursed: Decimal
    outstanding_principal: Decimal
    par_percentage: Decimal
    active_loans: int


class StaffPerformanceItemResponse(BaseModel):
    staff_id: int
    full_name: str
    role: str
    branch_name: str | None
    decisions_made: int
    approvals: int
    rejections: int

File: frontend/src/schemas/management.ts
// Mirrors backend app/schemas/management.py
export interface BranchRankingItemResponse {
  branch_id: number;
  branch_name: string;
  branch_code: string;
  total_disbursed: string;
  outstanding_principal: string;
  par_percentage: string;
  active_loans: number;
}

export interface StaffPerformanceItemResponse {
  staff_id: number;
  full_name: string;
  role: string;
  branch_name: string | null;
  decisions_made: number;
  approvals: number;
  rejections: number;
}
